src/main/java/com/oroarmor/multiitemlib/mixin/PlayerEntityMixin.java

```diff
@@ -24,19 +24,42 @@
 
 package com.oroarmor.multiitemlib.mixin;
 
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
+import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
+import com.oroarmor.multiitemlib.api.ICustomShieldDisableCooldown;
 import com.oroarmor.multiitemlib.api.UniqueItemRegistry;
+import net.minecraft.entity.player.ItemCooldownManager;
+import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.Redirect;
 
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 
 @Mixin(PlayerEntity.class)
 public class PlayerEntityMixin {
-    @Redirect(method = "damageShield(F)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isOf(Lnet/minecraft/item/Item;)Z"))
-    private boolean shields(ItemStack stack, Item isOfITem) {
-        return UniqueItemRegistry.SHIELD.isItemInRegistry(stack.getItem());
+    @Shadow @Final private ItemCooldownManager itemCooldownManager;
+
+    @WrapOperation(method = "damageShield(F)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isOf(Lnet/minecraft/item/Item;)Z"))
+    private boolean handleShieldDamage(ItemStack instance, Item item, Operation<Boolean> original) {
+        return UniqueItemRegistry.SHIELD.isItemInRegistry(instance.getItem());
+    }
+
+    @WrapOperation(method = "disableShield", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/player/ItemCooldownManager;set(Lnet/minecraft/item/Item;I)V"))
+    private void handleDisableShield(ItemCooldownManager instance, Item item, int duration, Operation<Void> original) {
+        for(Item registryEntry : UniqueItemRegistry.SHIELD.getValues()) {
+            int disableTime = duration;
+            if(registryEntry instanceof ICustomShieldDisableCooldown) {
+                disableTime = ((ICustomShieldDisableCooldown) registryEntry).getShieldCooldown();
+            }
+            original.call(instance, registryEntry, disableTime);
+        }
```

Comment:
This is wrong.
```suggestion
        if (item instanceof ICustomShieldDisableCooldown cooldown) {
                duration = cooldown.getShieldCooldown();
        }
        
        original.call(instance, item, duration);
```

Reply:
This specific change was done so you can't just swap it out with a normal shield to ignore cooldowns.

When you disable a shield, it disables everyone in your inventory, so I needed that behavior to apply to the netherite shield as well, so when yours is disabled it disables the normal shield as well

Reply:
Ah ok that makes sense. try this instead then:
```suggestion
        if (item instanceof ICustomShieldDisableCooldown cooldown) {
                duration = cooldown.getShieldCooldown();
        }
        
        for (Item shield : UniqueItemRegistry.SHIELD.getValues()) {
                original.call(instance, shield, duration);
        }
```

Reply:
That won't work either, that will just apply the custom shield cooldown to all shields. We need to do it the way I did so that way it applies the correct cooldowns to the correct Shields

Reply:
Yeah hm, there really isnt a good way of doing this. Ill need to think about it more when I write a more in depth API later.

Reply:
Yeah it's the way it is now because this gives the correct cooldown to each shield. So for example, your first suggested change would only disable the type of shield the player is holding and not other types of shields, the second would disable all shields in the inventory but give the player the cooldown of the shield they are holding, if it has a custom cooldown.